corehq/form_processor/abstract_models.py

```diff
@@ -37,6 +37,10 @@ def set_submission_properties(self, submission_post):
     def to_json(self):
         raise NotImplementedError()
 
+    @property
+    def is_normal(self):
```

Comment:
I don't think that's a bad idea. Rather than having individual fields for `is_error`, `is_archived` etc. you have one field that knows the state. Basically the same as `doc_type` in the JSON model. Could use a `PositiveSmallIntegerField`